src/model/bpmn/edge/Edge.ts

```diff
@@ -25,5 +26,6 @@ export default class Edge {
     readonly waypoints?: Waypoint[],
     readonly label?: Label,
     readonly messageVisibleKind: MessageVisibleKind = MessageVisibleKind.NONE,
+    readonly associationDirectionKind: AssociationDirectionKind = AssociationDirectionKind.NONE,
```

Comment:
According the BPMN specification (and the e2e test), the **associationDirectionKind** field is not on the **Edge**, but on the **Flow**.
So, I would rather have created a new **AssociationFlow** class, extended **Flow**, with this field.

<img width="713" alt="image" src="https://user-images.githubusercontent.com/4921914/87533302-b1e11880-c694-11ea-93b2-b58266e7c076.png">

Reply:
in fact this is how I did it - there is AssociationFlow that extends Flow

there was however redundant field in edge - I have cleared it as you are correct it should be enough to have it in AssociationFlow